core/src/com/mygdx/game/client/service/LocalTurnService.java

```diff
@@ -0,0 +1,39 @@
+package com.mygdx.game.client.service;
+
+import lombok.extern.java.Log;
+
+import java.util.UUID;
+import java.util.function.Function;
+
+@Log
+public class LocalTurnService implements TurnService {
+
+  private TurnToken activeToken = null;
+
+  @Override
+  public TurnToken peekToken() {
+    log.info("peeked token");
+    return activeToken;
+  }
+
+  @Override
+  public void obtainToken(Function<TurnToken, Void> onSuccess, Function<Throwable, Void> onError) {
```

Comment:
Ten onError narazie jest nieużywany, nie przeszkadza to?

Reply:
No nie przeszkadza. Bo zamysł jest taki, że jak uda się obtainować token to dostaniemy go i będziemy mogli coś z nim zrobić, a jak go nie dostaniemy to dostaniemy errora w tym onError i to będzie asynchroniczne. 

Ale problemem jest teraz to, że zauważyłem, że pomieszałem to podejście z podejściem w SyncListenerze, gdzie jest klasa Result i tam już się patrzy po booleanie isSuccesful. Tylko tam z kolei jest problem, że nie mamy informacji co właściwie się zepsuło. Jakby to było robione przez rxJava, reactora, albo przez kotlinowe korutyny to nie miałbym takim zagwozdek lol. 

Ale nadal nie wiem jaki dać interfejs tak naprawdę. Może zamiast Function<blah, blah> zrobić jakiś własny i nazwać go jakoś tak, żeby mówił, że przyjmuje tylko jakąś wartość, wykonuje pracę i kończy, ale nie zwraca niczego?

Reply:
Więc Kuba zdecyduj, czy wolisz mieć callbacki (i jakie interfejsy tam dać) czy wolisz Result, który nie mówi co konkretnie się stało?

Reply:
Narazie nie mogę sobie wyobrazić konsekwencji obydwu tych podejść, z mojej aktualnej intuicji lepsze się wydają Callbacki ale na tym etapie nie mam w głowie żadnych pomysłów na interfejsy, to co zaproponowałeś czyli własny interfejs wydaje się w porządku, ale no tak jak mówie, ciężko mi cos powiedzieć w tej chwili. Jednak jeśli oczekujesz konkretnej decyzji to zrób własny interfejs który opisałeś  "nazwać go jakoś tak, żeby mówił, że przyjmuje tylko jakąś wartość, wykonuje pracę i kończy, ale nie zwraca niczego"